routers/routes/routes.go

```diff
@@ -1002,6 +1002,12 @@ func RegisterRoutes(m *macaron.Macaron) {
 		}
 	})
 
+	m.Get("/apple-touch-icon.png", func(ctx *context.Context) {
+		file := path.Join(setting.StaticRootPath, "public/img/apple-touch-icon.png")
```

Comment:
This is not right because the file maybe embedded into binary. You should use functions from `modules/public` to serve file.

Reply:
Changed it to a redirect. Ideally, it should be a "rewrite" but I'm not sure how I would do that. I would like to avoid having to set all the caching headers and such which seems to be the case if I use `public.Asset`.

Reply:
Open to suggestions on how to delegate the request directly to the static file hander.